packages/bot-engine/src/scratch/dbot.js

```diff
@@ -78,11 +86,10 @@ class DBot {
      * JavaScript code that's fed to the interpreter.
      */
     runBot() {
-        const { run_panel }  = ScratchStore.instance.root_store;
         const should_run_bot = this.before_run_funcs.every(func => !!func());
         
         if (!should_run_bot) {
-            run_panel.onStopButtonClick();
+            this.stopBot();
```

Comment:
We called the store here because of the listeners that it unbinds, is this resolved?

Reply:
@aaron-binary it should be handled the way binary bot was handling it, if you know any case that listener is not disposed and can be tested please add it to QA

Reply:
@aaron-binary we are listening to stop event and when its emitted we dispose all the listener, pls let me know if you think there is any other case?
![image](https://user-images.githubusercontent.com/3080372/72678834-e47fcf00-3abe-11ea-959a-decd72133c05.png)

Reply:
@aaron-binary QA  and i both tested it for this couldn't see anything, if we face it again we will fix it, in the engine.